### PR TITLE
Ephesus merge mainnet liberated

### PR DESCRIPTION
**Pull request by dobertRowneySr**
merge `mainnet-liberated` into `ephesus` #4556 

# Main points
- the runtime & node version should be correct, but the QN / cli versioning should be reviewed
- ~~I left a comment for the `update_curator_group_permissions` call which I was not sure where it should be enabled or not~~ update: I left the proposal disabled as we don't have integration tests for this yet
- `pallet_membership::create_member` relies on `root` as it was used for founding members. Hence it is not used anymore and I have removed the respective tests from the `full` scenario
- There are still outstading PR targeting `ephesus`: 
    - #4594
    - #4575 
    - #4441
    - #4584
    - #4586
    - #4569 (I will retarget this after this PR has been merged)
    - #4598 (I will retarget this after this PR has been merged)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202078489349362/1203892485426670) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by bedeho** [comment]
Who else should review this?

**Comment by mnaamani** [comment]
The two flows a failing consistently in the network integration tests:

```
channel payouts proposal: Failed
group budget: Failed
```

**Comment by dobertRowneySr** [comment]
> The two flows a failing consistently in the network integration tests:
> 
> ```
> channel payouts proposal: Failed
> group budget: Failed
> ```

In the channelclaim-* fixtures there where Council budget checks. They where concurrent to the budget update logic run at each new block, that's why they where failing. Budget checks would have been nice, but implementing them is time consuming.

The other error was due to inviting members being run concurrently to the upgrade Membership Group budget fixture (so you had 2 concurrent budget updates)

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -234,10 +234,9 @@ impl Contains<<Runtime as frame_system::Config>::Call> for CallFilter {
         match call {
             Call::Content(content::Call::<Runtime>::destroy_nft { .. }) => false,
             Call::Content(content::Call::<Runtime>::toggle_nft_limits { .. }) => false,
-            // REVIEW THIS CHANGE:
-            // Call::Content(content::Call::<Runtime>::update_curator_group_permissions {
-            //     ..
-            // }) => false,
+            Call::Content(content::Call::<Runtime>::update_curator_group_permissions {
+                ..
+            }) => false,
```
Comment (mnaamani) [inline runtime/src/lib.rs:239]: Why was this originally filtered?

File: utils/api-scripts/src/initialize-lead.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ const workingGroupModules = [
   'contentWorkingGroup',
```
Comment (mnaamani) [inline utils/api-scripts/src/initialize-lead.ts]: I think we can drop this entierly. It uses sudo. We forgot about it in mainnet-liberated branch

**Comment by mnaamani** [comment]
> In the channelclaim-* fixtures there where Council budget checks. They where concurrent to the budget update logic run at each new block, that's why they where failing. Budget checks would have been nice, but implementing them is time consuming.

I would have assumed these were working before. Why would they stop working in this merge?

**Comment by ignazio-bovo** [comment]
> > In the channelclaim-* fixtures there where Council budget checks. They where concurrent to the budget update logic run at each new block, that's why they where failing. Budget checks would have been nice, but implementing them is time consuming.
> 
> I would have assumed these were working before. Why would they stop working in this merge?

The failing assertion were not directly messed up by the periodical budget bookkeeping, but by the `inviteMember` job being run concurrently to the `channelProposal` one.

**Comment by dobertRowneySr** [comment]
> > In the channelclaim-* fixtures there where Council budget checks. They where concurrent to the budget update logic run at each new block, that's why they where failing. Budget checks would have been nice, but implementing them is time consuming.
> 
> I would have assumed these were working before. Why would they stop working in this merge?

``` typescript
  public async execute(): Promise<void> {
    this.expectedClaims = await getExpectedClaims(this.api, this.claimChannelRewardParams)
    const expectedClaimsSum = this.expectedClaims.reduce((sum, claim) => sum.add(new BN(claim)), new BN(0))
    const councilBudgetPre = await this.api.query.council.budget()
    await super.execute()
    const councilBudgetPost = await this.api.query.council.budget()
    assert.equal(councilBudgetPre.toString(), councilBudgetPost.add(expectedClaimsSum).toString())
  }
```
The execute function reads the council budget before and post execution. However there are multiple budget modifications due to the council budget bookkeeping logic (screenshot below). So the assertion checking only for the channel reward variation fails.

Screenshot 2023-02-07 at 12 02 39

I am investigating why this error has not appeared before, 
Edit: the execution block for the `claim_channel_reward` changes at each test run. So it is possible that previous tests were fortuitous
Edit2: I can confirm that these assertions fails on ephesus @ 2676a17390ce1ccba2dbd9c6514d97e0886a8da8 at the moment

**Comment by mnaamani** [comment]
Ok thanks for investigating that, it would be good to have such assertions as part of these fixtures but it is out of scope of this PR.

**Review by mnaamani — APPROVED** [review]

**Pull request merged**